src/utils/ChatUtils.ts

```diff
@@ -48,6 +75,29 @@ export function deserialize_chat(text: string, characters: Map<string, Character
   return [chat, chars];
 }
 
+export async function deserialize_custom_chars(text: string, ds: CustomDataSource): Promise<CustomCharacter[]> {
+  const obj = JSON.parse(text);
+  const chars = await ds.get_characters();
+  // ensure that json without "custom_chars" field can also be imported
+  if (obj.custom_chars == undefined) {
+    return [];
+  }
+
+  const customChars = (obj.custom_chars as any[]).map(ch => {
+      const char = new CustomCharacter(ds, ch.name, ch.img)
```

Comment:
Missing `;` on many lines. If you use VSCode or an IDE, let it format the document and add the missing semicolons.